src/main.rs

```diff
@@ -166,104 +137,28 @@ fn main() -> Result<()> {
     Ok(())
 }
 
-/// Use a shared target directory to reduce (but not eliminate) disk
-/// space usage and needless rebuilds.
-fn setup_cargo_config(opts: &Opts) -> Result<()> {
-    let dir = opts.working_dir.join(".cargo");
+/// Use a shared target directory to reduce (but not eliminate) disk space usage and needless
+/// rebuilds.
+fn setup_cargo_config(working_dir: &Path) -> Result<()> {
+    let dir = working_dir.join(".cargo");
     fs::create_dir_all(&dir)?;
 
-    let path = dir.join("config");
+    let path = dir.join("config.toml");
     let config = format!(
         "[build]
          target-dir = '{}/target'",
-        opts.working_dir.display(),
+        working_dir.display(),
     );
 
     fs::write(path, config).map_err(Into::into)
 }
 
-#[derive(Debug, Default, Deserialize, Serialize)]
-struct ReverseDependencies {
-    dependencies: Vec<Dependency>,
-    versions: Vec<Version>,
-    meta: Meta,
-}
-
-#[derive(Debug, Deserialize, Serialize)]
-struct Dependency {
-    id: u64,
-    version_id: u64,
-    req: String, // ^0.6.10
-    downloads: u64,
-
-    #[serde(flatten)]
-    _other: BTreeMap<String, serde_json::Value>,
-}
-
-#[derive(Debug, Deserialize, Serialize)]
-struct Version {
-    id: u64,
-    #[serde(rename = "crate")]
-    name: String,
-    num: String,
-    downloads: u64,
-
-    #[serde(flatten)]
-    _other: BTreeMap<String, serde_json::Value>,
-}
-
-#[derive(Debug, Default, Deserialize, Serialize)]
-struct Meta {
-    total: u64,
-
-    #[serde(flatten)]
-    _other: BTreeMap<String, serde_json::Value>,
-}
-
-#[derive(Debug, Serialize)]
-struct ReverseDependenciesQuery {
-    page: u64,
-    per_page: u64, // max of 100
-}
-
 /// Find all crates that use the target crate
-fn fetch_reverse_dependencies(opts: &Opts, client: &Client) -> Result<ReverseDependencies> {
+fn fetch_reverse_dependencies(opts: &Opts, client: &SyncClient) -> Result<ReverseDependencies> {
     let cache_file = opts.working_dir.join("reverse_dependencies.json");
-
     use_cached_file_or_else(cache_file, || {
```

Comment:
I had debated on moving away from these quick-and-dirty JSON files to something a bit more structured (e.g. a SQLite DB) which might also remove the need to convert to/from JSON at all.

Reply:
This change was mostly about just using crates_io_api instead of manually doing calling into the website, working from a db dump directly could also be a useful approach if this needed to be fast (at the cost of that initial startup)